os/drivers/lwnl/lwnl_evt_queue.c

```diff
@@ -183,39 +158,19 @@ int lwnl_add_event(lwnl_cb_status type, void *buffer)
 	evt->flink = NULL;
 	evt->refs = 0;
 
-	switch (type) {
-	case LWNL_STA_CONNECTED:
-	case LWNL_STA_CONNECT_FAILED:
-	case LWNL_STA_DISCONNECTED:
-	case LWNL_SOFTAP_STA_JOINED:
-	case LWNL_SOFTAP_STA_LEFT:
-	case LWNL_SCAN_FAILED:
-	{
-		evt->data.status = type;
-		evt->data.data = NULL;
-		evt->data.data_len = 0;
-		break;
-	}
-	case LWNL_SCAN_DONE:
-	{
-		// ToDo
-		char *output = NULL;
-		int res = _lwnl_copy_scan_info(&output, (trwifi_scan_list_s *)buffer);
-		if (res < 0) {
-			evt->data.status = LWNL_SCAN_FAILED;
-			evt->data.data = NULL;
-			evt->data.data_len = 0;
-			break;
+	evt->data.status = type;
+	if (buffer) {
+		char *output = kmm_malloc(buf_len);
+		if (!output) {
+			LWQ_ERR;
+			return -3;
```

Comment:
Some functions in this file return negative value like `-3, -2`, not defined error code.
How about making and using defined error code with enum?

Reply:
defining error code will improve readability. but it is not relate to this PR. So I'll modify it later.